### PR TITLE
Introduced Pakistan Subdivisions and Province-Specific Holidays

### PR DESCRIPTION
**Pull request by Rana-10**
**Proposed Change**

    Added subdivisions for Pakistan, including Punjab, Sindh, Khyber Pakhtunkhwa, Balochistan, Gilgit-Baltistan, Azad Jammu & Kashmir, and Islamabad.

1.  Implemented _populate_subdiv_holidays() to handle province-specific holidays.
2.  Added Mela Chiraghan (March 29) as a localized holiday for Lahore (Punjab).
3.  Ensured self.subdiv is properly checked before assigning regional holidays.
4.  Maintained backward compatibility while improving holiday data granularity.

**Type of Change**

1. Supported country/market holidays update (calendar discrepancy fix, localization)
2. New feature (new holidays functionality in general)

**Checklist**

1. Followed the contributing guidelines.

2. Successfully ran make check, ensuring all tests pass.

### CODE REVIEW

**Comment by KJhellico** [comment]
> * Followed the contributing guidelines.
> * Successfully ran make check, ensuring all tests pass.

@Rana-10, are you sure about that?

**Review by Rana-10 — COMMENTED** [review]
I fixed a minor indentation error in this commit

**Review by Rana-10 — COMMENTED** [review]
Handled an edge case in this commit

File: holidays/countries/pakistan.py [hunk 1]
```diff
@@ -89,6 +99,28 @@ def _populate_public_holidays(self):
         self._add_ashura_eve(name)
         self._add_ashura_day(name)
 
+        self._populate_subdiv_holidays()
```
Comment (PPsyrius) [inline holidays/countries/pakistan.py:102]: This is no longer added in in newer implementations.

Instead, convert them to `_populate_subdiv_[subdiv_code]_public_holidays` like in [macau.py](https://github.com/vacanza/holidays/blob/dev/holidays/countries/macau.py#L381)

**Review by Rana-10 — COMMENTED** [review]
Made changes in line 104

**Comment by PPsyrius** [comment]
Please consider checking out https://github.com/vacanza/holidays/blob/dev/CONTRIBUTING.md

**Comment by KJhellico** [comment]
PR that does not pass pre-commit is not worth reviewing.

**Comment by arkid15r** [comment]
Closing as abandoned.
